### PR TITLE
[ROCKETMQ-26]add new way to send messages in asynchronous way 

### PR DESCRIPTION
**Pull request by thisisbaozi**
Jira issue: https://issues.apache.org/jira/browse/ROCKETMQ-26

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultSendPromise.java [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+import org.apache.rocketmq.client.producer.SendFuture;
+import org.apache.rocketmq.client.producer.SendResult;
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by stone
+ */
```
Comment (zhouxinyu) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultSendPromise.java]: No author info please, please refer to [here](http://rocketmq.apache.org/docs/code-guidelines/)

**Comment by vongosling** [comment]
Thanks @thisisbaozi , we have introduced a new way for async behavior in the latest version. Could you give us some hint for this usage ?

**Comment by thisisbaozi** [comment]
@vongosling I have added a example for it

**Comment by vongosling** [comment]
Thanks @thisisbaozi . RocketMQ 4.1 will publish a new async way to deal with the same requirement.

**Pull request closed without merging**